swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/images/AmazonImageHandler.kt

```diff
@@ -155,9 +156,17 @@ class AmazonImageHandler(
 
     log.debug("checking references for {} resources. Parameters: {}", images.size, params)
 
-    images.forEach {
-      if (it.name == null || it.description == null) {
-        it.set(NAIVE_EXCLUSION, true) // exclude these with exclusions
+    images.forEach { image ->
+      if (image.name == null || image.description == null) {
+        image.set(NAIVE_EXCLUSION, true) // exclude these with exclusions
+      }
+
+      if (image.blockDeviceMappings != null){
+        image.blockDeviceMappings.forEach { blockDevice ->
+          if (blockDevice.ebs != null && blockDevice.ebs.snapshotId != null) {
+            usedSnapshotRepository.recordUse(blockDevice.ebs.snapshotId, "aws:${params.region}:${params.account}") //todo eb: is this value needed?
```

Comment:
For reference, does it make sense to record which image is linked to a snapshot? I personally don't see a huge value in doing so, but could be used if we ever want to create a correlation matrix for different resource types. Just a thought, not something required here.

Reply:
I think it's useful (usecase: when we delete an image we could do some sort of 'restore' with the snapshot eventually). 

That being said, I'd prefer to refactor later and build something with that specific purpose in mind. This is just meant to build a mapping because aws only stores the mapping of image -> snapshot, and we need to know whether the image is in use.

Reply:
Discussed offline. To summarized, there will be a PR addressing a general approach to cache references, avoiding the need for a `Foo` resource to need a `FooInUseRepository`. Ideally we would keep an`InUseRepository` to handle in use resource references for resources.